### PR TITLE
Integration with Galois field library

### PR DESCRIPTION
**Pull request by Acentelles**

### CODE REVIEW

**Review by sdiehl — APPROVED** [review]

**Comment by sdiehl** [comment]
Update the ChangeLog

File: Bulletproofs/ArithmeticCircuit/Internal.hs [hunk 1]
```diff
@@ -180,30 +182,7 @@ genIdenMatrix size = (\x -> (\y -> fromIntegral (fromEnum (x == y))) <$> [1..siz
...
-
-generateRandomAssignment :: forall f m . (Num f, AsInteger f, Crypto.MonadRandom m) => Integer -> m (Assignment f)
-generateRandomAssignment n = do
-  aL <- replicateM (fromIntegral n) ((fromInteger :: Integer -> f) <$> generateMax (2^n))
-  aR <- replicateM (fromIntegral n) ((fromInteger :: Integer -> f) <$> generateMax (2^n))
-  let aO = aL `hadamardp` aR
-  pure $ Assignment aL aR aO
-
-computeInputValues :: (Field f, Eq f) => GateWeights f -> [[f]] -> Assignment f -> [f] -> [f]
+computeInputValues :: (Fractional f, Eq f) => GateWeights f -> [[f]] -> Assignment f -> [f] -> [f]
```
Comment (Multramate) [inline Bulletproofs/ArithmeticCircuit/Internal.hs]: Replace this with `KnownNat p => PrimeField p`

File: Bulletproofs/ArithmeticCircuit/Internal.hs [hunk 2]
```diff
@@ -212,7 +191,7 @@ computeInputValues GateWeights{..} wV Assignment{..} cs
         ^+^ vectorMatrixProductT aO wO
         ^-^ cs
 
-gaussianReduce :: (Field f, Eq f) => [[f]] -> [[f]]
+gaussianReduce :: (Fractional f, Eq f) => [[f]] -> [[f]]
```
Comment (Multramate) [inline Bulletproofs/ArithmeticCircuit/Internal.hs]: Replace this with `KnownNat p => PrimeField p`

File: Bulletproofs/ArithmeticCircuit/Internal.hs [hunk 3]
```diff
@@ -247,13 +226,73 @@ gaussianReduce matrix = fixlastrow $ foldl reduceRow matrix [0..length matrix-1]
         nz = List.last (List.init row)
 
 -- Solve a matrix (must already be in REF form) by back substitution.
-substituteMatrix :: (Field f, Eq f) => [[f]] -> [f]
+substituteMatrix :: (Fractional f, Eq f) => [[f]] -> [f]
```
Comment (Multramate) [inline Bulletproofs/ArithmeticCircuit/Internal.hs]: Replace this with `KnownNat p => PrimeField p`

File: Bulletproofs/ArithmeticCircuit/Internal.hs [hunk 4]
```diff
@@ -247,13 +226,73 @@ gaussianReduce matrix = fixlastrow $ foldl reduceRow matrix [0..length matrix-1]
...
 substituteMatrix matrix = foldr next [List.last (List.last matrix)] (List.init matrix)
   where
     next row found = let
       subpart = List.init $ drop (length matrix - length found) row
       solution = List.last row - sum (zipWith (*) found subpart)
       in solution : found
 
-solveLinearSystem :: (Field f, Eq f) => [[f]] -> [f]
+solveLinearSystem :: (Fractional f, Eq f) => [[f]] -> [f]
```
Comment (Multramate) [inline Bulletproofs/ArithmeticCircuit/Internal.hs]: Replace this with `KnownNat p => PrimeField p`

**Review by Multramate — CHANGES_REQUESTED** [review]
Replace `Fractional f` with `KnownNat p => PrimeField p`

**Comment by sdiehl** [comment]
This should fix the CI.

**Review by Multramate — APPROVED** [review]

**Pull request merged**